### ISSUE TITLE
Serial No selection in Warranty Claim throws exception

### ISSUE THREAD

**Issue by TurkerTunali** [frappe/erpnext#38330]
### Information about bug

Serial No selection in the Warranty Claim throws below exception. 

Basically it is saying that, there is no Customer field in the Serial No.

https://github.com/frappe/erpnext/assets/710051/cc495133-a394-4542-bfe4-75fde0e899d3

### Module

support

### Version

ERPNext: v15.2.0 (version-15)
Frappe Framework: v15.1.0 (version-15)

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
### App Versions

{
	"erpnext": "15.2.0",
	"frappe": "15.1.0",
	"print_designer": "1.0.0"
}
```
### Route
```
Form/Warranty Claim/new-warranty-claim-dgxfijylui
```
### Traceback
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 110, in application
    response = frappe.api.handle(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
           ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1683, in call
    return fn(*args, **newargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/desk/search.py", line 47, in search_link
    results = search_widget(
              ^^^^^^^^^^^^^^
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/desk/search.py", line 194, in search_widget
    values = frappe.get_list(
             ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1949, in get_list
    return frappe.model.db_query.DatabaseQuery(doctype).execute(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/db_query.py", line 200, in execute
    result = self.build_and_run()
             ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/db_query.py", line 240, in build_and_run
    return frappe.db.sql(
           ^^^^^^^^^^^^^^
  File "apps/frappe/frappe/database/database.py", line 217, in sql
    self._cursor.execute(query, values)
  File "env/lib/python3.11/site-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 558, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 822, in _read_query_result
    result.read()
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 1200, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "env/lib/python3.11/site-packages/pymysql/connections.py", line 772, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.11/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.11/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1054, "Unknown column 'tabSerial No.customer' in 'where clause'")

```
### Request Data
```
{
	"type": "POST",
	"args": {
		"txt": "",
		"doctype": "Serial No",
		"ignore_user_permissions": 0,
		"reference_doctype": "Warranty Claim",
		"filters": "[[\"Serial No\",\"docstatus\",\"!=\",2],[\"Serial No\",\"customer\",\"=\",\"Deneme-Test\"]]"
	},
	"headers": {},
	"error_handlers": {},
	"url": "/api/method/frappe.desk.search.search_link",
	"request_id": null
}
```
### Response Data
```
{
	"exception": "pymysql.err.OperationalError: (1054, \"Unknown column 'tabSerial No.customer' in 'where clause'\")",
	"exc_type": "OperationalError"
}
```
```

**Comment by nutcracker53** [comment]
Hello, I am still encountering this issue in my erpnext v15 . Do I need to re-install v15 to get the fix? Thanks

**Comment by s-aga-r** [comment]
> Hello, I am still encountering this issue in my erpnext v15 . Do I need to re-install v15 to get the fix? Thanks

The fix is released in v14.53.0 and v15.5.0, update your instance accordingly.
